strawberryfields/backends/bosonicbackend/backend.py

```diff
@@ -28,13 +28,28 @@
 
 from strawberryfields.backends.bosonicbackend.bosoniccircuit import BosonicModes
 from strawberryfields.backends.base import NotApplicableError
+from strawberryfields.program_utils import CircuitError
+import sympy
 
 
 def kron_list(l):
     """Take Kronecker products of a list of lists."""
     return reduce(np.kron, l)
 
 
+def parameter_checker(parameters):
+    """Checks if any items in an iterable are sympy objects."""
+    for item in parameters:
+        if isinstance(item, sympy.Expr):
+            return True
+
+        # This checks all the nested items if item is an iterable
+        if hasattr(item, "__iter__") and not isinstance(item, str):
```

Comment:
(minor) Perhaps `isinstance(item, Iterable)`, from `from collections.abc import Iterable` is preferred here? 🤔 It requires an extra import, but might be slightly clearer.

Reply:
Done!